hypothesis-python/src/hypothesis/extra/lark.py

```diff
@@ -119,7 +125,8 @@ def __init__(self, grammar, start, explicit, alphabet):
             )
         self.terminal_strategies.update(explicit)
 
-        nonterminals = {}
+        # TODO: should this be NonTerminal only? but rule.expansion is Symbol...
+        nonterminals: dict[str, list[tuple[Symbol, ...]]] = {}
```

Comment:
The `rules` returned by `grammar.compile()` are all nonterminals, the components of which are therefore a list of sequences of any kind of symbols (terminals, constant strings, and other nonterminals).  So this annotation is correct, although the variable names are not very good 😓